### DISCUSSION TITLE
Back to formulize, and I'm unable to update or make fresh install

### DISCUSSION THREAD

**Discussion by flaggada** [jegelstaff/formulize#519] [Tech Support] [answered]
Hello Julian and everybody

Long time since I didn't use formulize. I was working with formulize 5.0, before I Ieft computer.

Now with new mysql version I'm trying to bring my website to life.
So I've made a fresh install of latest xoops release, mysql 8.1, and old fashioned formulize 5.0. After that I copy my oldest xoops database of formulize. But it doesn't work fine...

Then I tried to put the 7.3 version and it gave me some error like "Error : Error: Undefined constant "SDATA_DB_NAME" "

I tried to apply the database patch, same problem.
I' also tried to make a fresh install for impresscms with the package, but It's unable to create some table at the end of the process.

I'm a bit lost. Can you help me?

**Comment by jegelstaff** [comment]
Hello, great to see you again :)

The SDATA error message is probably related to a difference in your old mainfile.php or trust path file, versus what is typically there now. That constant is one of the ones that's defined in the trust path file normally and that controls access to the database. If that's not defined then you would certainly get an error and nothing else would work.

If you are doing this on top of the latest XOOPS then you would need to be sure to use only the modules/formulize folder from the 7.3 release. If you use all the files then you are not using XOOPS anymore. What we release is an entire working system not just the module. 

If you are trying to put Formulize 7.3 into XOOPS, I am not sure what the results would be, I haven't run it in XOOPS for a long time but we have tried very hard to make it compatible with the old XOOPS APIs, so if the current version still has xoops_getmodulehandler and all that old stuff, then I don't see why it wouldn't work in theory but might need some touching up around the edges. 

I would be very happy to take a look at the system and get it running. I would need access to the files on the server though and maybe that's not simple to arrange. I suggest joining the slack channel and we can chat about it there in more detail: https://join.slack.com/t/formulize/shared_invite/zt-1f3qfgcip-fSdTuE4LEBfEfefd2S~EUg

There's been so many amazing things since version 5 it would be really awesome to get that all in place for you!

Take it easy,
--Julian

**Comment by jegelstaff** [comment]
Hello,

So, the issue with the installer and the tables not being created, is resolved in commit e58961b4fad6fa8ce8a8f62d3757bdcf4ebd3a39

The problem only exists with MariaDB version 10 and later, which introduced a function called MID. The use of that term in the SQL for creating tables, needed to be avoided/escaped. Unfortunately, this suggests there may be other issues in the SQL elsewhere, because if there are things that use the term mid in a way that acts and looks like a function, then they would fail as well.

I don't think that's the case though. This problem was also sort of caused by really old MySQL syntax that used the old term KEY instead of INDEX. The lines in question were:

```
KEY mid (mid)
```

Which is kind of crazy. Now it is:

```
INDEX(`mid`)
```

I'm pretty sure there won't be other references in the system to the term `mid` that are used in this way.

--Julian

**Comment by jegelstaff** [comment, marked as answer]
Another reason I don't think this is a serious issue in general is that we routinely use/test with MariaDB 11. That's the LTS version we use with Docker, I believe.

I suspect this issue has happened for a long time, but we would have always just ignored it because those tables that it couldn't create, have absolutely no effect on Formulize whatsoever. But it is better now that they are being created! Thank you!

--Julian
